### ISSUE TITLE
 [feature request]how to merge many models to one model with shared backbone just use some code ,not a create a new model

### ISSUE THREAD

**Issue by designerZhou** [pytorch/pytorch#70244]
I train some models with different datas ,these models' some parameters are shared ,
when i inference the models ,i need merge the models to one model ,i know the shared op ,so ,i want to merge these models
shared op to one op  with seperate head only when inference not train.
    i don't want to write a new model ,could i write a function to merge the op with same dict name ,and create a new inference model 
automatic .
   or maybe any way else is ok 
  thank you !

**Comment by samdow** [comment]
If you're using PyTorch modules, you can edit the forward function to change based on whether the training flag is set or not. An example of this in use is the [BatchNorm module](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/batchnorm.py#L145-L152)

Closing this because in general, we use github issues only for bug reports or feature requests. Please use the forum to ask questions: https://discuss.pytorch.org/. If using the training flag does not fit your use case, please try asking over there.
